lib/libyui_client/filter.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+module LibyuiClient
+  class Filter
+    attr_reader :full, :plain, :regex
+
+    def initialize(filter)
+      @full = build_filters(filter)
+      @plain = @full.reject { |_, v| v.is_a? Regexp }
+      @regex = @full.select { |_, v| v.is_a? Regexp }
+    end
+
+    def to_s
+      @full.to_s
+    end
+
+    private
+
+    def build_filters(hash)
+      allowed_filters = %i[id label type class]
+      filter = hash.select do |key, _|
+        key if allowed_filters.include?(key.to_sym)
```

Comment:
I've just noticed that we are not returning any error now when it is called from ouside, app.<control> like before.